datadog-log/src/writers.rs

```diff
@@ -1,13 +1,193 @@
 // Copyright 2025-Present Datadog, Inc. https://www.datadoghq.com/
 // SPDX-License-Identifier: Apache-2.0
 
+use crate::logger::FileConfig;
 use crate::logger::StdTarget;
+use chrono;
 use std::io::Write;
-use std::path::Path;
+use std::path::{Path, PathBuf};
 use std::{fs, io};
 use tracing_appender::non_blocking::{NonBlocking, WorkerGuard};
 use tracing_subscriber::fmt::MakeWriter;
 
+/// A custom file appender that handles optional size-based rotation
+/// tokio doesn't support file rotation, so we need to implement it ourselves.
+/// https://github.com/tokio-rs/tracing/pull/2497
+struct CustomFileAppender {
+    path: PathBuf,
+    current_size: u64,
+    max_size: u64,
+    max_files: u64,
+    current_file: fs::File,
+}
+
+impl CustomFileAppender {
+    fn new(config: &FileConfig) -> io::Result<Self> {
+        let path = Path::new(&config.path).to_path_buf();
+        let file = fs::OpenOptions::new()
+            .create(true)
+            .append(true)
+            .open(&path)?;
+
+        let current_size = file.metadata()?.len();
+
+        Ok(Self {
+            path,
+            current_size,
+            max_size: config.max_size_bytes,
+            max_files: config.max_files,
+            current_file: file,
+        })
+    }
+
+    fn get_timestamp_string() -> String {
+        chrono::Local::now().format("%Y-%m-%d_%H-%M-%S").to_string()
```

Comment:
Does this potentially need to be more granular than just a second? Thinking about the edge cases where someone sets a very small max size and / or if the log level is set to something verbose. This should be a rare situation, so non-blocking if we want to potentially handle in a follow-up.